micro_dl/networks/base_unet.py

```diff
@@ -50,16 +50,17 @@ def __init__(self, network_config, predict=False):
                 network_config['height'] // (2 ** num_down_blocks)
             assert feature_height_at_last_block >= 2, \
                 'network depth is incompatible with height'
-
         #  keras upsampling repeats the rows and columns in data. leads to
         #  checkerboard in upsampled images. repeat - use keras builtin
         #  nearest_neighbor, bilinear: interpolate using custom layers
-        upsampling = network_config['upsampling']
+        upsampling = self.config['upsampling']
         msg = 'invalid upsampling, not in repeat/bilinear/nearest_neighbor'
         assert upsampling in ['bilinear', 'nearest_neighbor', 'repeat'], msg
 
-        self.config = network_config
-        if 'depth' in network_config and self.config['depth'] > 1:
+        if 'depth' in self.config:
+            if not predict:
+                assert self.config['depth'] > 1, \
```

Comment:
Can depth not be 1 for a 2D model?

Reply:
Good question! Hmm... not really sure. I was thinking that depth will be None or not provided for a 2D model. Modified to a 2D model when depth = 1

Reply:
Thank you. It makes sense to keep >1 in stack_2D, but in base_unet it can be good to accept depth =1 (at least I've put it to 1 when running 2D from not knowing better).